src/mqt/predictor/ml/predictor.py

```diff
@@ -275,8 +277,53 @@ def train_random_forest_classifier(
         Returns:
             True when the training was successful, False otherwise.
         """
-        training_data = self.get_prepared_training_data()
+        training_data = Predictor.get_prepared_training_data(self.figure_of_merit)
+        clf = self._train_random_forest_model(training_data, None, self.figure_of_merit, save_classifier)
```

Comment:
I feels a bit wrong to me that here, the device defaults to "None". Maybe also an aspect to discuss in person.